src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +359,510 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
-                                         onValueChanged={this.onValueChanged}
-                                         vm={vm} />}
-                    </Form>
-                    <AdditionalOptions cacheMode={this.state.cacheMode}
-                                       device={this.state.device}
-                                       idPrefix={idPrefix}
-                                       onValueChanged={this.onValueChanged}
-                                       busType={this.state.busType}
-                                       serial={this.state.serial}
-                                       validationFailed={validationFailed}
-                                       supportedDiskBusTypes={this.props.supportedDiskBusTypes} />
-                </>
-            );
-        }
+    };
 
-        return (
-            <Modal position="top" variant="medium" id={`${idPrefix}-dialog-modal-window`} isOpen onClose={Dialogs.close}
-                   title={isMediaInsertion ? _("Insert disc media") : _("Add disk")}
-                   footer={
-                       <>
-                           <Button id={`${idPrefix}-dialog-add`}
-                                   variant='primary'
-                                   isLoading={this.state.addDiskInProgress || this.state.verificationInProgress}
-                                   isDisabled={this.state.addDiskInProgress || this.state.verificationInProgress || dialogLoading ||
-                                               (storagePools.length == 0 && this.state.mode != CUSTOM_PATH) ||
-                                               (this.state.mode == CUSTOM_PATH && this.state.device === "disk" && this.state.customDiskVerificationFailed)}
-                                   onClick={isMediaInsertion ? this.onInsertClicked : this.onAddClicked}>
-                               {isMediaInsertion ? _("Insert") : _("Add")}
-                           </Button>
-                           <Button id={`${idPrefix}-dialog-cancel`} variant='link' onClick={Dialogs.close}>
-                               {_("Cancel")}
-                           </Button>
-                       </>
-                   }>
-                {this.state.dialogError && <ModalError dialogError={this.state.dialogError} dialogErrorDetail={this.state.dialogErrorDetail} />}
-                {defaultBody}
-            </Modal>
-        );
-    }
-}
+    return (
+        <>
+            <Button id={`${idPrefix}-dialog-add`}
+                    variant='primary'
+                    isLoading={addDiskInProgress || verificationInProgress}
+                    isDisabled={addDiskInProgress || verificationInProgress || dialogLoading ||
+                                (storagePools.length == 0 && mode != CUSTOM_PATH) ||
+                                (validate && Object.keys(validationFailed).length > 0)}
+                    onClick={onClick}>
+                {isMediaInsertion ? _("Insert") : _("Add")}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231114-134752-e3097bd0-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)